engine/swarm/services.md

```diff
@@ -264,24 +264,50 @@ Swarm mode lets you network services in a couple of ways:
 * publish ports externally to the swarm using ingress networking
 * connect services and tasks within the swarm using overlay networks
 
-### Publish ports externally to the swarm
+### Publish ports
 
-You publish service ports externally to the swarm using the `--publish
-<TARGET-PORT>:<SERVICE-PORT>` flag. When you publish a service port, the swarm
-makes the service accessible at the target port on every node regardless if
-there is a task for the service running on the node.
+When you create a swarm service, you can publish that service's ports to hosts
+outside the swarm in two ways:
 
-For example, imagine you want to deploy a 3-replica nginx service to a 10-node
-swarm as follows:
+- [You can rely on the routing mesh](#publish-a services-ports-using-the-routing-mesh).
+  When you publish a service port, the swarm makes the service accessible at the
+  target port on every node, regardless of whether there is a task for the
+  service running on that node or not. This is less complex but creates some
+  additional networking overhead.
+
+- [You can publish a service task's port directly on the swarm node](#publish-a-services-ports-directly-on-the-swarm-node)
+  where that service is running. This feature is available in Docker 1.13 and
+  higher. This creates less networking overhead by passing the routing mesh, but
+  you are responsible for routing requests to your service's nodes, and you can
+  only run one task for a given service on a given swarm node.
+
+Keep reading for more information and use cases for each of these methods.
+
+#### Publish a service's ports using the routing mesh
+
+To publish a service's ports externally to the swarm, use the `--publish
+<TARGET-PORT>:<SERVICE-PORT>` flag. The swarm
+makes the service accessible at the target port **on every swarm node**. If an
+external host connects to that port on any swarm node, the routing mesh routes
+it to a task. The external host does not need to know the IP addresses or
+internally-used ports of the service tasks to interact with the service.
+
+##### Example: Run a three-task Nginx service on 10-node swarm
+
+Imagine that you have a 10-node swarm, and you deploy an Nginx service running
+three tasks on a 10-node swarm:
 
 ```bash
-docker service create --name my_web --replicas 3 --publish 8080:80 nginx
+$ docker service create --name my_web \
+                        --replicas 3 \
+                        --publish 8080:80 \
+                        nginx
 ```
 
-The scheduler will deploy nginx tasks to a maximum of 3 nodes. However, the
-swarm makes nginx port 80 from the task container accessible at port 8080 on any
-node in the swarm. You can direct `curl` at port 8080 of any node in the swarm
-to access the web server:
+Three tasks will run on up to three nodes. Multiple tasks can run on the same
```

Comment:
> Multiple tasks can run on the same node.

While this is possible, Swarmkit scheduler tries to spread the tasks to different nodes as long as it can. This sentence may be a little misleading.